### ISSUE TITLE
Add the function method `plot!(plot, data::Result)`

### ISSUE THREAD

**Issue by oameye** [NonlinearOscillations/HarmonicBalance.jl#59]
Add a method to the HarmonicBalance Plot function where you can add data to plot by the notation `plot!(plot, data::Result)` just as in the normal `Plots.jl` syntax.

![image](https://user-images.githubusercontent.com/57623933/195369678-3cdae4f2-d183-45e2-aec2-be8abd020053.png)

**Comment by oameye** [comment]
I am not sure how we can achieve this with multiple dispatch. However, if we would follow the guidelines of Plots.jl to make a recipe instead of using multiple dispatch it has this feature build in automatically.  I also suggested this in #67.

Indeed, if we use the following example of the documentation
```julia
using Plots

mutable struct MyType end
 @recipe function f(::MyType, n::Integer = 10; add_marker = false)
           linecolor   --> :blue
           seriestype  :=  :path
           markershape --> (add_marker ? :circle : :none)
           delete!(plotattributes, :add_marker)
           rand(n)
end
```
Then writing
```julia
mt1 = MyType(); mt2 = MyType()
p = plot(mt1)
plot!(p, mt2)
```
creates
![image](https://user-images.githubusercontent.com/57623933/197423957-7bff512a-4fcb-41d4-b2fd-4f3ef76554b7.png)

**Comment by jkosata** [comment]
To be solved by #67
